cats/cats-sql/src/main/kotlin/com/netflix/spinnaker/cats/sql/cache/SqlCache.kt

```diff
@@ -893,57 +1022,115 @@ class SqlCache(
     }
   }
 
-  private fun getHash(body: String?): String? {
+  /**
+   * Returns a hash of the body given if not null or blank, otherwise returns null.
+   *
+   * @param body the body to hash.
+   * @return hash of the body.
+   */
+  private fun getBodyHash(body: String?): String? {
     if (body.isNullOrBlank()) {
       return null
     }
-    return try {
-      val digest = MessageDigest.getInstance("SHA-256")
-        .digest(body.toByteArray())
-      digest.fold("") { str, it ->
-        str + "%02x".format(it)
-      }
-    } catch (e: Exception) {
-      log.error("error calculating hash for body: $body", e)
-      null
-    }
+    return getSha256Hash(body)
+  }
+
+  /**
+   * Returns a hash of the id given.
+   *
+   * @param id The id to hash.
+   * @return hash of the id.
+   */
+  private fun getIdHash(id: String): String {
+    return getSha256Hash(id)
+  }
+
+  /**
+   * Returns a hash of the agent given.
+   *
+   * @param agent the agent to hash.
+   * @return hash of the agent.
+   */
+  private fun getAgentHash(agent: String): String {
```

Comment:
Is this a separate method from `getIdHash(id: String)` just in case someday we want to hash agents differently?